File: C/product_of_digits.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    long long n;
    printf("Enter a number: ");
    if (scanf("%lld", &n) != 1) return 0;

    long long x = llabs(n);
    long long prod = 1;

    if (x == 0) prod = 0;
    else {
        while (x > 0) {
            prod *= (x % 10);
            x /= 10;
        }
    }

    printf("Product of digits: %lld\n", prod);
    return 0;
}

File: Python/product_of_digits.py
def main():
    n = int(input("Enter a number: "))
    n = abs(n)

    prod = 1
    if n == 0:
        prod = 0
    else:
        while n > 0:
            prod *= n % 10
            n //= 10

    print("Product of digits:", prod)


if __name__ == "__main__":
    main()

